src/linked_list/linked_list.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations
+
+from src.linked_list.linked_list_node import LinkedListNode
+from typing import Any, List
+
+
+class LinkedList:
+    def __init__(self, initial: Any = None) -> None:
+        self.head = None
+        self.tail = None
+        self.length = 0
+
+        if initial:
+            node = LinkedListNode(initial)
+
+            self.head = node
+            self.tail = node
+            self.length = 1
+
+    def append(self, value: Any) -> LinkedList:
+        node = LinkedListNode(value)
+
+        if not self.head:
+            self.head = node
+            self.tail = node
+        else:
+            self.tail.next = node
+            self.tail = node
+        self.length += 1
+
+        return self
+
+    def prepend(self, value: Any) -> LinkedList:
+        node = LinkedListNode(value)
+
+        if not self.head:
+            self.head = node
+            self.tail = node
+        else:
+            node.next = self.head
+            self.head = node
+        self.length += 1
+
+        return self
+
+    def __str__(self, separator: str = " -> ") -> str:
```

Comment:
Replace it with `def __str__(self) -> str:
        values = [str(node.data) for node in self.to_list()]
        return ' -> '.join(values)`
        because it's more readable.